lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -72,7 +72,7 @@ def self.deliver(*args)
       automate_attrs = options[:attrs].dup
 
       if object_type
-        vmdb_object = object_type.constantize.find_by(:id => object_id)
+        vmdb_object = object_type.constantize.find_by!(:id => object_id)
```

Comment:
Very happy to see this.  Can we get a test around this change?  Might want to consider break this if-block out into a smaller method.  The `deliver_queue ` method is scary big.  :scream:

Reply:
I have asked @pkomanek to look into refactoring this method so we can move forward with this change.